docs/data/date-pickers/date-calendar/WeekPicker.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import isBetweenPlugin from 'dayjs/plugin/isBetween';
+import { styled } from '@mui/material/styles';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateCalendar } from '@mui/x-date-pickers/DateCalendar';
+import { PickersDay, PickersDayProps } from '@mui/x-date-pickers/PickersDay';
+
+dayjs.extend(isBetweenPlugin);
+
+interface CustomPickerDayProps extends PickersDayProps<Dayjs> {
+  isSelected: boolean;
+  isHovered: boolean;
+}
+
+const CustomPickersDay = styled(PickersDay, {
+  shouldForwardProp: (prop) => prop !== 'isSelected' && prop !== 'isHovered',
+})<CustomPickerDayProps>(({ theme, isSelected, isHovered, day }) => ({
+  ...((isSelected || isHovered) && {
+    borderRadius: 0,
+  }),
+  ...(isSelected && {
+    backgroundColor: theme.palette.primary.main,
+    color: theme.palette.common.white,
+    '&:hover, &:focus': {
+      backgroundColor: theme.palette.primary.main,
+    },
+  }),
+  ...(isHovered && {
+    backgroundColor: theme.palette.primary.light,
+    '&:hover, &:focus': {
+      backgroundColor: theme.palette.primary.light,
```

Comment:
https://github.com/mui/mui-x/assets/3309670/7c2d8e0e-6068-4c98-a97c-721bc92bce1a

Not sure why but if I add a transition I end up with this weird double transition effect.

Reply:
It's because the `transition` on the `PickersDay` component is defined incorrectly.
It should be always defined, not wrapped within a `selected` rule. 🙈